contrib_bots/lib/converter.py

```diff
@@ -0,0 +1,235 @@
+# See readme.md for instructions on running this code.
+
+from __future__ import division
+from past.utils import old_div
+
+import copy
+from math import log10, floor
+
+# A dictionary allowing the conversion of each unit to its base unit.
+# An entry consists of the unit's name, a constant number and a constant
+# factor that need to be added and multiplied to convert the unit into
+# the base unit in the last parameter.
+UNITS = {'acre': [0, 4046.8564224, 'square-meter'],
+         'are': [0, 100, 'square-meter'],
+         'bit': [0, 1, 'bit'],
+         'byte': [0, 8, 'bit'],
+         'celsius': [273.15, 1, 'kelvin'],
+         'cubic-centimeter': [0, 0.000001, 'cubic-meter'],
+         'cubic-decimeter': [0, 0.001, 'cubic-meter'],
+         'cubic-foot': [0, 0.028316846592, 'cubic-meter'],
+         'cubic-inch': [0, 0.000016387064, 'cubic-meter'],
+         'cubic-meter': [0, 1, 'cubic-meter'],
+         'cubic-yard': [0, 0.764554857984, 'cubic-meter'],
+         'fahrenheit': [255.372222, 0.555555, 'kelvin'],
+         'fluid-ounce': [0, 0.000029574, 'cubic-meter'],
+         'foot': [0, 0.3048, 'meter'],
+         'gram': [0, 1, 'gram'],
+         'hectare': [0, 10000, 'square-meter'],
+         'inch': [0, 0.0254, 'meter'],
+         'kelvin': [0, 1, 'kelvin'],
+         'liter': [0, 0.001, 'cubic-meter'],
+         'meter': [0, 1, 'meter'],
+         'mile': [0, 1609.344, 'meter'],
+         'nautical-mile': [0, 1852, 'meter'],
+         'ounce': [0, 28.349523125, 'gram'],
+         'pound': [0, 453.59237, 'gram'],
+         'square-centimeter': [0, 0.0001, 'square-meter'],
+         'square-foot': [0, 0.09290304, 'square-meter'],
+         'square-inch': [0, 0.00064516, 'square-meter'],
+         'square-kilometer': [0, 1000000, 'square-meter'],
+         'square-meter': [0, 1, 'square-meter'],
+         'square-mile': [0, 2589988.110336, 'square-meter'],
+         'square-yard': [0, 0.83612736, 'square-meter'],
+         'ton': [0, 1000000, 'gram'],
+         'yard': [0, 0.9144, 'meter']}
+
+PREFIXES = {'atto': -18,
+            'femto': -15,
+            'pico': -12,
+            'nano': -9,
+            'micro': -6,
+            'milli': -3,
+            'centi': -2,
+            'deci': -1,
+            'deca': 1,
+            'hecto': 2,
+            'kilo': 3,
+            'mega': 6,
+            'giga': 9,
+            'tera': 12,
+            'peta': 15,
+            'exa': 18}
+
+ALIASES = {'c': 'celsius',
```

Comment:
I can sort the aliases alphabetically by the alias though, right now they are sorted alphabetically by the unit's name.